src/languages.ts

```diff
@@ -35,11 +35,32 @@ export class Languages {
   private loadExtensionMap = () => {
     const extensions: ExtensionsTypes = {};
 
+    /**
+     * The extension map can contain multiple languages with the same extension,
+     * but we only want a single one. For the moment, these clashes are resolved
+     * by the simple heuristic below listing high-priority languages. We may want
+     * to consider smarter heuristics to correctly identify languages in cases
+     * where the extension is ambiguous. The ordering of the list matters and
+     * languages earlier on will get a higher priority when resolving clashes.
+     */
+    const importantLanguages = ["javascript", "typescript", "ruby", "python", "java", "c", "c++", "c#", "rust", "scala", "perl", "go"];
```

Comment:
Could you move this out to a top-level constant?

Reply:
And just curious, can you give an example of where the ordering here is important?

Reply:
The only example I found where the ordering _actually_ mattered was the case of `.spec` that I described in the PR description as this extension appears in a few languages but (to the best of my understanding) is _mostly_ important for Ruby where it is common in unit tests. I still ordered the other languages roughly by popularity in case any ambiguities arise in their extensions in future, but at the moment there are no other ambiguities among these languages (except `.h` being shared between `c` and `c++` but in CodeQL we don't care about that because they both map to the same thing).